lib/plug/ssl.ex

```diff
@@ -58,6 +61,40 @@ defmodule Plug.SSL do
   require Logger
   import Plug.Conn
 
+  @strong_tls_ciphers [
+    'ECDHE-RSA-AES256-GCM-SHA384',
+    'ECDHE-ECDSA-AES256-GCM-SHA384',
+    'ECDHE-RSA-AES128-GCM-SHA256',
+    'ECDHE-ECDSA-AES128-GCM-SHA256',
+    'DHE-RSA-AES256-GCM-SHA384',
+    'DHE-RSA-AES128-GCM-SHA256'
+  ]
+
+  @compatible_tls_ciphers [
+    'ECDHE-RSA-AES256-GCM-SHA384',
+    'ECDHE-ECDSA-AES256-GCM-SHA384',
+    'ECDHE-RSA-AES128-GCM-SHA256',
+    'ECDHE-ECDSA-AES128-GCM-SHA256',
+    'DHE-RSA-AES256-GCM-SHA384',
+    'DHE-RSA-AES128-GCM-SHA256',
+    'ECDHE-RSA-AES256-SHA384',
+    'ECDHE-ECDSA-AES256-SHA384',
+    'ECDHE-RSA-AES128-SHA256',
+    'ECDHE-ECDSA-AES128-SHA256',
+    'ECDHE-RSA-AES256-SHA',
+    'ECDHE-ECDSA-AES256-SHA',
+    'DHE-RSA-AES256-SHA256',
+    'DHE-RSA-AES128-SHA256',
+    'ECDHE-RSA-AES128-SHA',
+    'ECDHE-ECDSA-AES128-SHA'
```

Comment:
The order of these last 6 seems inconsistent: either all the DHE- ciphers should be at the end, or all the -SHA ones, depending on what is prioritised. Looking at the OWASP list it looks like they consider the SHA1 ciphers a step down in security (which on paper it is, though a collision attack is currently only practical for SHA1 certificates, not for the SHA1 used in TLS frame MACs, so that doesn't really apply here). So the DHE- ciphers with -SHA256 should move up two lines...?

Reply:
Agreed. Good catch. I meant to do that.